http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -0,0 +1,202 @@
+              logger.info(s"in update of handleClusterTransitionFinished, followup machine config: ${followupMachineConfig}, follow-up masterMachineType: ${followupMachineConfig.get.masterMachineType} followup map: ${followupMap}")
+              //if there is a record of a follow-up being needed, perform the follow-up
+              if (!followupMachineConfig.isEmpty && !followupMachineConfig.get.masterMachineType.isEmpty) {
+                for {
+                  // perform gddao and db updates for new resources
+                  _ <- clusterHelper.updateMasterMachineType(resolvedCluster, MachineType(followupMachineConfig.get.masterMachineType.get))
+                  x = logger.info("in for comprehension for handleClusterTransitionFinished")
+                  // start cluster
+                  _ <- clusterHelper.internalStartCluster(resolvedCluster)
+                } yield ()
+              } else {
+                IO.raiseError(new WorkbenchException(s"Recieved message ${message} that a cluster finished stopping and wishes to update, " +
+                  s"but proper followup details (aka machineConfig) were not saved in the subscriber at the time of the initial action. Saved config: ${followupMachineConfig}"))
+              }
+            }
+            case None => IO.raiseError(ClusterNotFoundException(message.clusterFollowupDetails.clusterId))
+          }
+        } yield result
+      }
+
+      //No actions for other statuses yet. There is some logic that will be needed for all other cases (i.e. the 'None' case where no cluster is found in the db and possibly the case that checks for a key in the followupMap.
+      // TODO: Refactor once there is more than one case
+      case _ => {
+        logger.info(s"received a message notifying that cluster ${message.clusterFollowupDetails.clusterId} has transitioned to status ${message.clusterFollowupDetails.clusterStatus}. This is a Noop.")
+        IO.unit
+      }
+    }
+
+    resolution.unsafeToFuture().failed
+      .foreach { e =>
+        logger.error(s"Error occurred updating cluster with id ${message.clusterFollowupDetails.clusterId} after stopping", e)
+      }
+
+    resolution
+  }
+
+}
+sealed trait LeoPubsubMessage {
+  def messageType: String
+}
+
+final case class ClusterFollowupDetails(clusterId: Long, clusterStatus: ClusterStatus) extends Product with Serializable
+
+abstract class ClusterTransitionRequestedMessage(updatedMachineConfig: MachineConfig, clusterId: Long)  extends LeoPubsubMessage
+
+final case class StopUpdateMessage(updatedMachineConfig: MachineConfig, clusterId: Long)
+  extends ClusterTransitionRequestedMessage(updatedMachineConfig: MachineConfig, clusterId: Long) {
+  val messageType = "stopUpdate"
+}
+
+case class ClusterTransitionFinishedMessage(clusterFollowupDetails: ClusterFollowupDetails) extends LeoPubsubMessage {
+  val messageType = "transitionFinished"
+}
+
```

Comment:
Can we simplify the hierarchy a bit and just have  `StopUpdateMessage` and `ClusterTransitionFinishedMessage` extends `LeoPubsubMessage` directly? I don't know if `sealed` will work as intended when there're extra layers of inheritance